src/Library.php

```diff
@@ -49,24 +49,19 @@ public static function define_tables() {
 	 * Set up the database tables which the plugin needs to function.
 	 */
 	public static function maybe_create_tables() {
-		$db_version = get_option( 'wc_blocks_db_version', 0 );
-
-		if ( version_compare( $db_version, \Automattic\WooCommerce\Blocks\Package::get_version(), '>=' ) ) {
-			return;
-		}
-
 		global $wpdb;
 
-		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+		$schema_version    = 260;
```

Comment:
Is this a hardcoded number that will get changed anytime we have a schema update? Should we document this somewhere in the repo docs? Would there be value in having this as a part of the `Package` configuration (so we have one class that's responsible for all version information - including db version)?

Reply:
I didn't want to tie this to version so it would run less frequently. Since DB related things live here, maybe best to keep the schema version here too?

It should be bumped when making a change to the dB or creating new tables.

Reply:
> I didn't want to tie this to version so it would run less frequently. 

Right I agree you don't want to tie to the plugin version, but I still think it's valuable to keep anything to do with versioning in a single class/file so there's a canonical place to go for that information. Alternatively, could have a designated DB service class that contains the schema and version info (but that might be overkill for the plugin). I'm just wary of `Library` being the place where we throw in all plugin initialization type things.

Reply:
Wherever it goes it should be kept with the schema so it's not forgotten. It's all consumed here and has little to do with the Package. 

I'm also not sure how permanent this code is; when this table and class is merged to core, we no longer need it (once minimum support is bumped). This and the ReserveStock class can be removed. 

We could alternatively just track the install of this table specifically and remove the version system with that in mind?

Reply:
Let's just keep it here for now then but keep in mind:

- I think we should keep the version system, it's unknown if we'll have other scenarios where a new table is needed and versioning is the easiest to work with for those cases (less code churn).
- I'm still wary about dumping things like this into Library. I agree that we'll be removing this once the table is in core and minimum required version is bumped so it's likely not too much of a worry here (as long as we are pro-active about removing redundant code).